app/src/main/java/com/google/android/sambadocumentsprovider/browsing/SmbServer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.google.android.sambadocumentsprovider.browsing;
+
+import android.net.Uri;
+
+public interface SmbServer {
```

Comment:
Actually since we know we're gonna make some changes to accommodate mDNS later. Maybe we can remove this interface, unless you think it can still be useful later on.

I don't want to remove this interface in follow-up changes. It only indicates we didn't put thorough thoughts into this problem.